User.BusinessLogic/Extensions/Cache.cs

```
@@ -17,17 +19,19 @@ internal static class Cache
         return data;
     }
 
-    public static async Task CacheData<T>(this IDistributedCache distributedCache, T data, TimeSpan cacheLifetime, string key, CancellationToken cancellationToken)
+    public static async Task CacheData<T>(this IDistributedCache distributedCache, T data, string key, CancellationToken cancellationToken)
     {
         if (data is null)
             return;
 
+        if (!int.TryParse(lifetime, out var lifetimeMinutes))
+            return;
```

Don't you think to throw an exception instead of returning null?